lms/static/js/student_account/enrollment.js

```diff
@@ -52,7 +63,54 @@
                     }
                 });
             },
+            /**
+             * Show a message in the frontend.
+             * @param  {Object} message The message to display.
+             */
+            showMessage: function(message) {
+                const componentId = 'student-enrollment-feedback-error';
+                const existing = document.getElementById(componentId);
+                if (existing) {
+                    existing.remove();
+                }
+                // Using a fixed dashboard URL as the redirect destination since this is the most logical
+                // place for users to go after encountering an enrollment error. The URL is hardcoded
+                // because environment variables are not injected into the HTML/JavaScript context.
+                const DASHBOARD_URL = '/dashboard';
+                const textContent = (message && message.detail) ? message.detail : String(message);
+                const messageDiv = document.createElement('div');
+                messageDiv.setAttribute('id', componentId);
+                messageDiv.setAttribute('class', 'fixed-top d-flex justify-content-center align-items-center');
+                messageDiv.style.cssText = [
+                    'width:100vw',
+                    'height:100vh',
+                    'background:rgba(0,0,0,0.5)',
+                    'z-index:9999'
+                ].join(';');
 
+                const buttonText = typeof gettext === 'function' ? gettext('Close') : 'Close';
+
+                messageDiv.innerHTML = `
+                  <div class="page-banner w-75 has-actions">
+                    <div class="alert alert-warning" role="alert">
+                      <div class="row">
```

Comment:
I suggest using `w-100`, so the row occupies 100% of its parent div.
```suggestion
                      <div class="row w-100">
```

Screenshot before:
![image](https://github.com/user-attachments/assets/93b9397f-ab9a-46a3-b2c8-6f32a9b4e231)

After:
![image](https://github.com/user-attachments/assets/b27003a9-8ca3-4b11-95ed-eb76d153845c)

Before, on small screens:
![image](https://github.com/user-attachments/assets/51698024-1b85-4b32-b79c-99a21e778536)

After:
![image](https://github.com/user-attachments/assets/9eefb267-d7c4-4d85-96eb-68d7290bdd8a)

Reply:
Done!